backend/src/Nebulift.Api/Program.cs

```diff
@@ -1,36 +1,52 @@
-namespace Nebulift.Api;
-
-/// <summary>
-/// The main program class.
-/// </summary>
-public static class Program
-{
-    /// <summary>
-    /// The main entry point for the application.
-    /// </summary>
-    /// <param name="args">The command-line arguments.</param>
-    public static void Main(string[] args)
-    {
-        var builder = WebApplication.CreateBuilder(args);
-
-        // Add services to the container.
-        builder.Services.AddControllers();
-        builder.Services.AddEndpointsApiExplorer();
-        builder.Services.AddSwaggerGen();
-
-        var app = builder.Build();
-
-        // Configure the HTTP request pipeline.
-        if (app.Environment.IsDevelopment())
-        {
-            app.UseSwagger();
-            app.UseSwaggerUI();
-        }
-
-        app.UseHttpsRedirection();
-        app.UseAuthorization();
-        app.MapControllers();
-
-        app.Run();
-    }
+namespace Nebulift.Api;
+using Nebulift.Api.Templates;
+using EnvironmentName = Microsoft.Extensions.Hosting.EnvironmentName;
+
+public static class Program
+{
+    public static void Main(string[] args)
+    {
+        var builder = WebApplication.CreateBuilder(args);
+
+        // Register the string (templates folder path) in DI
+        builder.Services.AddSingleton("../../../templates/");

```

Comment:
Register configurations using the [options pattern](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-9.0) (register using `builder.Services.Configure` and inject `IOptions` in the service) instead of injecting a single string as it will be more clear and more scalable if we need to add other options later

Reply:
Ok done